KAA-1481: 1-thread callback executor (Java client) causes issue with "nested" callback calls

Serega290696 opened the pull request.

vilimets committed:
KAA-1402 Obj-C Endpoint access token update flow fix

rasendubi committed:
KAA-1457 Link mbedtls unconditionally

Commit:
OJI-1392: Code changes to support cross-compilation for QNX 6.5. Add installation instructions.

Commit:
Merge branch 'release-0.10' of https://github.com/kaaproject/kaa into feature/KAA-1392-new

Commit:
KAA-1392: Add docs to an assembly.

Commit:
KAA-1136: Using Kaa endpoint SDKs, C - reviewed.

Commit:
KAA-1138: How to contribute - reviewed.

Commit:
KAA-1487: Git flow - reviewed.

Commit:
KAA-1403: Fix access token setter

Commit:
KSF-88: Getting started - fixed upon developer review.

Serega290696 committed:
Fix KAA-1481

Serega290696 committed:
Fix KAA-1481

Serega290696 committed:
Unit test for FlexibleExecutorContext and refactor FlexibleExecutorContext

Serega290696 committed:
Fix KAA-1481

Serega290696 committed:
Fix KAA-1481

Serega290696 committed:
Unit test for FlexibleExecutorContext and refactor FlexibleExecutorContext

Serega290696 committed:
Unit test for FlexibleExecutorContext

Serega290696 committed:
Unit test for FlexibleExecutorContext

Serega290696 committed:
Merge remote-tracking branch 'beltser/feature/KAA-1481' into feature/KAA-1481

# Conflicts:
#	client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java

vadimol committed:
Merge pull request #1149 from hunch7/kaa-1136-EPsdk-c-review

KAA-1136: Using Kaa endpoint SDKs, C - reviewed.

vadimol committed:
Merge pull request #1134 from dyosick/feature/KAA-1392

KAA-1392: [C++ SDK] QNX 6.5.0 support.

Commit:
KSF-88: Getting started - fixed upon developer review.

Bridged adapter mode mentioned while maintaining NAT as the default mode.

Commit:
KAA-1136: Using endpoint SDK, C - fixed upon developer review.

Commit:
KAA-1487: Git flow - fixed upon developer review.

Commit:
KAA-1138: How to contribute - fixed upon developer review.

vadimol committed:
Merge pull request #1167 from hunch7/kaa-1136-EPsdk-c-fix

KAA-1136: Using endpoint SDK, C - fixed upon developer review.

Rib47 commented on client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java:
```diff
@@ -0,0 +1,211 @@
+
+
+        public FlexibleExecutorContext build() {
+            FlexibleExecutorContext f = new FlexibleExecutorContext(this);
```
You must check builder fields for **null** values. If any **null** value will be found you must throw an exception or  set fields to default values. The second alternative is better one.

vadimol committed:
KAA-1498 Removed link to nonexistent page

Commit:
KAA-1179: Single node installation - reviewed.

Commit:
KAA-1160: Glossary - fixed upon AK review.

rasendubi committed:
Merge pull request #1156 from hunch7/kaa-1487-gitflow-review

KAA-1487: Git flow - reviewed.

rasendubi committed:
Merge pull request #1154 from hunch7/kaa-1138-howtocontribute-review

KAA-1138: How to contribute - reviewed.

vadimol committed:
Merge pull request #1159 from GamovCoder/KAA-1403

KAA-1403: Fix access token setter

rasendubi committed:
Merge pull request #1169 from vadimol/KAA-1498

KAA-1498 Removed link to nonexistent page

Commit:
KSF-88: Getting started - fixed upon developer review.

Commit:
KAA-1160: Glossary - 'Kaa extension' definition added.

GeorgePostolakiy committed:
KAA-702: Merge_conflict_fix

vadimol committed:
Synchronized avrogen.sh template

vadimol committed:
Fixed cmake bug with avrogen

vadimol committed:
Merge pull request #1173 from kaaproject/vadimol-patch-1

Fixed cmake bug with avrogen

vadimol committed:
Merge pull request #1172 from GeorgePostolakiy/KAA-702

KAA-702: Merge_conflict_fix

vadimol committed:
Merge pull request #1128 from rasendubi/KAA-1457

KAA-1457 Link mbedtls unconditionally

rasendubi committed:
Merge pull request #1161 from hunch7/ksf-88-gstarted-fix

KSF-88: Getting started - fixed upon developer review.

rasendubi committed:
Merge pull request #1088 from vilimets/KAA-1402

KAA-1402 [Objective-C] Endpoint access token is not sent to Kaa cluster

ohamov committed:
KAA-1403 Ignore error codes

The functions can be called while the channels are not
initialized. It's permitted and the error codes should be
ignored in this case.

Slash32 committed:
APP-353: Synchronized template

Also added NOTE to avoid modification of the auto-generated file.

vadimol committed:
Merge pull request #1176 from Slash32/patch-1

APP-353: Synchronized template file

Serega290696 committed:
Fix KAA-1481

Serega290696 committed:
Fix KAA-1481

Commit:
Merge remote-tracking branch 'beltser/feature/KAA-1481' into feature/KAA-1481

Conflicts:
	client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java

Serega290696 committed:
Merge remote-tracking branch 'beltser/feature/KAA-1481' into feature/KAA-1481

Conflicts:
	client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java

Rib47 commented on client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java:
```diff
@@ -0,0 +1,185 @@
+package org.kaaproject.kaa.client.context;
```
missed license in this file
see documentation:
https://docs.google.com/document/d/1gesi7pElMnoT5ocefV0qLooEMnZuTvTc81xSdJQGGTI

Rib47 commented on client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java:
```diff
@@ -0,0 +1,185 @@
+package org.kaaproject.kaa.client.context;
+
```
missed license in this file
see documentation:
https://docs.google.com/document/d/1gesi7pElMnoT5ocefV0qLooEMnZuTvTc81xSdJQGGTI

Rib47 reviewed (changes requested):
Missed licenses

Commit:
Merge remote-tracking branch 'beltser/feature/KAA-1481' into feature/KAA-1481

Commit:
KAA-1160: Glossary - refined upon AK review.

akokhanovskyi committed:
Merge pull request #1171 from hunch7/kaa-1160-glossary-reviewed

KAA-1160: Glossary - fixed upon AK review.

vadimol committed:
Merge pull request #1170 from hunch7/kaa-1179-singlenodeinstallation-review

KAA-1179: Single node installation - reviewed.

vadimol committed:
Merge pull request #1175 from GamovCoder/KAA-1403

KAA-1403 Ignore error codes

Commit:
KAA-1160: Glossary links fixed

vadimol committed:
Merge pull request #1178 from hunch7/GlossaryLinksFix

KAA-1160: Glossary links fixed

Commit:
KAA-1481: 1-thread callback executor (Java client) causes issue with "nested" callback calls

Serega290696 committed:
Merge remote-tracking branch 'beltser/feature/KAA-1481' into feature/KAA-1481

Conflicts:
	client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java

Serega290696 committed:
Merge branch 'feature/KAA-1481' of https://github.com/Serega290696/kaa into feature/KAA-1481

# Conflicts:
#	client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java
#	client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java
